test_utils/src/mock_signing.rs

```diff
@@ -0,0 +1,99 @@
+use holochain_core_types::{
+    agent::{AgentId, KeyBuffer},
+    cas::content::{Address, AddressableContent},
+};
+use holochain_dpki::keypair::{Keypair, SEEDSIZE, SIGNATURESIZE};
+use holochain_sodium::secbuf::SecBuf;
+use jsonrpc_ws_server::jsonrpc_core::{self, types::params::Params, IoHandler};
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+
+lazy_static! {
+    pub static ref TEST_AGENT_KEYS: Mutex<HashMap<Address, Arc<Mutex<Keypair>>>>
+        = Mutex::new(HashMap::new());
+}
```

Comment:
What do you think about making this a `Mutex<Vec<_>>` Instead of `Mutex<HashMap<_>>`, and then `registered_test_agent` doesn't take a nick and instead appends a new agent with name `"test-agent-N"`, to guarantee that we don't have overlap between tests?

Reply:
I don’t think it’s bad to have overlap in the agent IDs. But getting different agent IDs depending on the ordering of simultaneously running tests would be bad because the ID would not be deterministic and we couldn’t have fixtures for signatures and addresses.